manager/controllers/app/plotter_generator.go

```diff
@@ -181,18 +184,32 @@ func (p *PlotterGenerator) addInMemoryStep(element *ResolvedEdge, datasetID stri
 	if lastStepAPI == nil {
 		assetID = datasetID
 	}
-	steps = append(steps, app.DataFlowStep{
-		Cluster:  element.Cluster,
-		Template: string(element.Module.Spec.Capabilities[element.CapabilityIndex].Capability),
-		Parameters: &app.StepParameters{
-			Source: &app.StepSource{
-				AssetID: assetID,
-				API:     lastStepAPI,
+	if flowType == taxonomy.WriteFlow {
+		steps = append(steps, app.DataFlowStep{
+			Cluster:  element.Cluster,
+			Template: string(element.Module.Spec.Capabilities[element.CapabilityIndex].Capability),
+			Parameters: &app.StepParameters{
+				Sink: &app.StepSink{
```

Comment:
Why do we need sink?

Reply:
In the current plotter generated the  write flow contains the asset as a source rather than sink: (also mentioned in the issue)
`- assetId: fybrik-notebook-sample/new-data-parquet flowType: write name: fybrik-notebook-sample/new-data-parquet-write subFlows: - flowType: write name: "" steps: - - cluster: thegreendragon name: "" parameters: api: connection: fybrik-arrow-flight: hostname: my-notebook-fybrik-notebook-sample-arrow-flight-aef23.fybrik-blueprints port: 80 scheme: grpc name: fybrik-arrow-flight source: assetId: fybrik-notebook-sample/new-data-parquet template: write triggers: - workload```